src/context/useProducts.js

```diff
@@ -0,0 +1,15 @@
+import { createContext } from "react";
+import { useReducer,useContext } from "react";
+import { dataReducer } from "../reducers/dataReducer";
+const ProductsContext=createContext(null)
+
+export const useProducts=()=>useContext(ProductsContext);
+
+export const ProductsProvider=({children})=>{
+    const [state,dispatch]=useReducer(dataReducer,{categories:[],products:[],filteredProducts:[],sortByPrice:'',filterByRatings:'',categoryActionFigures:null,categoryMasks:null,categoryTshirts:null,isProducts:false,slider:0})
```

Comment:
suggestion: U can extract this part in seperate variable as initial value

Reply:
Okay I will do it. Thanks for the suggestion.